src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -1294,6 +1305,65 @@ private bool OnlyWhitespaceOrCommentsAfterExtent(InternalScriptExtent extent)
             return true;
         }
 
+        internal bool IsPipeContinuance(IScriptExtent extent)
+        {
+            var scriptExtent = (InternalScriptExtent)extent;
+            return scriptExtent.EndOffset < _script.Length && PipeContinuanceAfterExtent(scriptExtent);
+        }
+
+        private bool PipeContinuanceAfterExtent(InternalScriptExtent extent)
+        {
+            // If the first non-newline, non-whitespace, non-comment, non-backtick, non-semi-colon character
+            // following the newline is a pipe, we have a pipe continuance.
+            for (int i = extent.EndOffset; i < _script.Length;)
+            {
+                if (_script[i] == '#')
+                {
+                    // SkipLineComment will return the position after the comment end
+                    // which is either at the end of the file, or a cr or lf.
+                    i = SkipLineComment(i + 1);
+                    continue;
+                }
+
+                if (_script[i] == '<' && (i + 1) < _script.Length && _script[i + 1] == '#')
+                {
+                    i = SkipBlockComment(i + 2);
+                    continue;
+                }
+
+                if (_script[i].IsWhitespace())
+                {
+                    i++;
+                    continue;
+                }
+
+                if (_script[i] == '\n' || _script[i] == '\r')
+                {
+                    i = SkipNewline(i);
+                    continue;
+                }
+
+                if (_script[i] == '`' && (i + 1) < _script.Length)
+                {
+                    if (_script[i + 1] == '\n' || _script[i + 1] == '\r')
+                    {
+                        i = SkipNewline(i + 1);
```

Comment:
The same (double check).

Reply:
I inlined the skip to avoid additional redundant checks.